payment/src/on_failed/main.py

```diff
@@ -76,3 +78,21 @@ def handler(event, _):
     payment_token = get_payment_token(order_id)
     cancel_payment(payment_token)
     delete_payment_token(order_id)
+
+    # Generate custom metrics using the Embedded Metric Format
+    # See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html
+    print(json.dumps({
+        "paymentCanceled": 1,
+        "environment": ENVIRONMENT,
+        "_aws": {
+            # Timestamp is in milliseconds
+            "Timestamp": int(datetime.datetime.now().timestamp()*1000),
+            "CloudWatchMetrics": [{
+                "Namespace": "ecommerce.payment",
+                "Dimensions": [["environment"]],
+                "Metrics": [
+                    {"Name": "paymentCanceled"}
+                ]
+            }]
+        }
+    }))
```

Comment:
__Typo__: This should be `Cancelled`.
```suggestion
    print(json.dumps({
        "paymentCancelled": 1,
        "environment": ENVIRONMENT,
        "_aws": {
            # Timestamp is in milliseconds
            "Timestamp": int(datetime.datetime.now().timestamp()*1000),
            "CloudWatchMetrics": [{
                "Namespace": "ecommerce.payment",
                "Dimensions": [["environment"]],
                "Metrics": [
                    {"Name": "paymentCancelled"}
                ]
            }]
        }
    }))
```